services/instagram_service/controller/instagram_controller.py

```diff
@@ -0,0 +1,26 @@
+import asyncio
+from services.instagram_service.src import client_grpc
+from services.instagram_service.src import InstagramService
+class instagram_Contoller:
+    def __int__(self, instagram_service: InstagramService, client: client_grpc) -> None:
+        self.client = client
+        self.instagram_service = instagram_service
+
+    def read_direct_messages(self) -> list:
+        return self.instagram_service.read_direct_messages()
+
+    def run(self, query: str) -> str:
+        return self.client.run(query)
+
+    def send_direct_messages(self, message: str, sender_username: str) -> None:
+        return self.instagram_service.send_direct_message(message, sender_username)
+
+    async def start(self, interval_seconds=60) -> None:
+        while True:
+            direct_messages = self.read_direct_messages()
+            if direct_messages:
```

Comment:
Consider improving the readability of your code by the use of the negative condition. A more Pythonic way to write this would be:

```python
if direct_messages is None:
    continue
```
This makes the code more intuitive and easier to read.